### PR TITLE
doc: minor sg docs tweaks, link to hack hour

### PR DESCRIPTION
**Pull request by bobheadxi**
- Minor wording tweaks from a read-through of the docs
- Remove long-term vision section since some are done and this is better tracked in issues
- Makes the hack hour official: https://github.com/sourcegraph/handbook/pull/1711
- Development tips, including note on https://github.com/sourcegraph/sourcegraph/issues/29222#issuecomment-1007558458

(my sg hack hour contribution 😛 )

### CODE REVIEW

File: doc/dev/background-information/sg/vision.md [hunk 1]
```diff
@@ -19,17 +19,6 @@ You can also watch [this video](https://drive.google.com/file/d/1DXjjf1YXr8Od8vG
...
+## Contributing
 
-The following are ideas for what could/should be built into `sg`:
-
-- Create `sg setup` command that sets up local environment, including dependencies https://github.com/sourcegraph/sourcegraph/issues/24900
-- Replace every shell script in `./dev` with an `sg` command
-  - Replace `./dev/generate.sh` with an `sg generate` command https://github.com/sourcegraph/sourcegraph/issues/25441
-  - Replace `./dev/drop-entire-local-database-and-redis.sh` with an `sg` command
-  - ...
-- Get rid of the `dev-private` repository and handle shared site-configuration, credentials, and licenses in `sg`.
```
Comment (jhchabran) [inline doc/dev/background-information/sg/vision.md:31]: We should create an issue for this one I think.

File: doc/dev/background-information/sg/vision.md [hunk 2]
```diff
@@ -19,17 +19,6 @@ You can also watch [this video](https://drive.google.com/file/d/1DXjjf1YXr8Od8vG
...
-- Build log handling into `sg` so it's easier to debug what's happening in a local environment. https://github.com/sourcegraph/sourcegraph/issues/25442
-- Add `sg generate-graphql-resolver-and-stub-methods` command that has a better name but creates all the boilerplate needed to create new GraphQL resolvers in Go
```
Comment (jhchabran) [inline doc/dev/background-information/sg/vision.md:33]: Another one, could be the beginning of skeletons and templates to speed things up 😇

File: doc/dev/background-information/sg/vision.md [hunk 3]
```diff
@@ -19,17 +19,6 @@ You can also watch [this video](https://drive.google.com/file/d/1DXjjf1YXr8Od8vG
 - Stripe's `pay` command, [described here](https://buttondown.email/nelhage/archive/papers-i-love-gg/)
 - [Stack Exchange Local Environment Setup](https://twitter.com/nick_craver/status/1375871107773956103?s=21) command
 
-## Long term
```
Comment (mrnugget) [inline doc/dev/background-information/sg/vision.md:22]: I agree that the list is outdated and probably not the best format, but I do think that it's valuable to have a list of things that tells a potential contributor "oh, this is all in scope for this project & they also see these things [i.e. dev-private] as a problem"

**Review by mrnugget — APPROVED** [review]

File: doc/dev/background-information/sg/vision.md (hunk 1 above)
Comment (bobheadxi) [reply doc/dev/background-information/sg/vision.md:31]: I can't seem to find it, so created one: https://github.com/sourcegraph/sourcegraph/issues/29640

File: doc/dev/background-information/sg/vision.md (hunk 2 above)
Comment (bobheadxi) [reply doc/dev/background-information/sg/vision.md:33]: https://github.com/sourcegraph/sourcegraph/issues/29641

File: doc/dev/background-information/sg/vision.md (hunk 3 above)
Comment (bobheadxi) [reply doc/dev/background-information/sg/vision.md:22]: I added a link to `sg` issues here!

**Pull request merged**